### PR TITLE
check for available http port

### PR DESCRIPTION
**Pull request by mattkasun**
## Describe your changes
netclient daemon gets available port for http server
port is saved to config file
netclient-gui uses config file to set http endpoint

## Provide testing steps
see NET-29
## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [x] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [x] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [x] My unit tests pass locally.
- [ ] Netclient & Netmaker are awesome.

### CODE REVIEW

**Comment by afeiszli** [comment]
Just a general comment, but this is a lot of changes for a relatively small problem. Can you explain why so many files need to be touched for this?

**Comment by Aceix** [comment]
FreeBSD build is broken, static check fails

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

**Comment by mattkasun** [comment]
> 

most of the changes are to ensure that when /etc/netclient is created, it is created with the correct permissions

File: functions/httpserver.go [hunk 1]
```diff
@@ -20,12 +22,22 @@ type Network struct {
 
 func HttpServer(ctx context.Context, wg *sync.WaitGroup) {
 	defer wg.Done()
+	port, err := ncutils.GetFreeTCPPort(8000)
```
[the remark this answers is not in the document]
Comment (mattkasun) [reply functions/httpserver.go]: simplified GetFreeTCPPort

**Comment by afeiszli** [comment]
> > 
> 
> most of the changes are to ensure that when /etc/netclient is created, it is created with the correct permissions

shouldn't that be a separate PR, or does that relate to the port issue?

**Comment by mattkasun** [comment]
It relates to port issue .  Permissions need to be correct or the unprivileged Gui would not be able to read it's config file

**Review by Aceix — APPROVED** [review]

**Review by afeiszli — APPROVED** [review]

**Review by theguy951357 — APPROVED** [review]
pr steps pass
critical list passes

**Pull request merged**